tests/ui/test_process_graph_generator.py

```diff
@@ -0,0 +1,78 @@
+"""
+Unit tests for ProcessGraphGenerator.
+
+Author: Kevin Fu
+"""
+
+import os
+import tempfile
+import unittest
+
+from gtsfm.ui.process_graph_generator import ProcessGraphGenerator
+
+# needed to test registry properly
+from tests.ui.test_gtsfm_process import FakeImageLoader, FakeOutputGTSFM  # noqa: F401
+
+
+class TestProcessGraphGenerator(unittest.TestCase):
+    def test_save_graph(self):
+        """Ensure ProcessGraphGenerator saves graph as png/svg files without crashing."""
+
+        process_graph_generator = ProcessGraphGenerator(test_mode=True)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            png_path = os.path.join(tempdir, "output.png")
+            svg_path = os.path.join(tempdir, "output.svg")
+            extra_dir_path = os.path.join(tempdir, "output", "output.png")
+
+            process_graph_generator.save_graph(png_path)
+            process_graph_generator.save_graph(svg_path)
+            process_graph_generator.save_graph(extra_dir_path)
+
+            # check that files were saved, not their correctness
+            self.assertTrue(os.path.exists(png_path))
+            self.assertTrue(os.path.exists(svg_path))
+            self.assertTrue(os.path.exists(extra_dir_path))
+
+    def test_no_fake_nodes(self):
+        """
+        Ensure that no test nodes from test_registry.py are in the final
```

Comment:
nit: check [style guide](https://github.com/borglab/gtsfm/blob/master/CONTRIBUTING.md). docstrings should have a 1-line description on the same line as the triple quote.

Can add additional info below that, separated by a line.